// bootstrap-vue-next/dist/index-CY1kFvDO.mjs
import { _ as _sfc_main, a as _sfc_main$1 } from "./BAccordion.vue_vue_type_script_setup_true_lang-Dc3-1efg.mjs";
import { _ as _sfc_main$2 } from "./BAlert.vue_vue_type_script_setup_true_lang-CWJ2tlQa.mjs";
import { _ as _sfc_main$3, a as _sfc_main$4 } from "./BAvatarGroup.vue_vue_type_script_setup_true_lang-xF00GQDT.mjs";
import { _ as _sfc_main$5 } from "./BBadge.vue_vue_type_script_setup_true_lang-Dy7CqA47.mjs";
import { _ as _sfc_main$6, a as _sfc_main$7 } from "./BBreadcrumb.vue_vue_type_script_setup_true_lang-aab8VaEA.mjs";
import { _ as _sfc_main$8 } from "./BButton.vue_vue_type_script_setup_true_lang-D-y5ASo-.mjs";
import { _ as _sfc_main$9, a as _sfc_main$a } from "./BButtonToolbar.vue_vue_type_script_setup_true_lang-BSwHiGXg.mjs";
import { _ as _sfc_main$m } from "./BCloseButton.vue_vue_type_script_setup_true_lang-DCcXsu_x.mjs";
import { _ as _sfc_main$b, a as _sfc_main$c, b as _sfc_main$d, c as _sfc_main$f, d as _sfc_main$g, e as _sfc_main$h, f as _sfc_main$j } from "./BCard.vue_vue_type_script_setup_true_lang-B87L_9wi.mjs";
import { _ as _sfc_main$e, a as _sfc_main$i } from "./BCardText.vue_vue_type_script_setup_true_lang-DdqzUfNr.mjs";
import { _ as _sfc_main$k, a as _sfc_main$l } from "./BCarouselSlide.vue_vue_type_script_setup_true_lang-ByU4s8W1.mjs";
import { _ as _sfc_main$o } from "./BCollapse.vue_vue_type_script_setup_true_lang-ByE-TAjy.mjs";
import { _ as _sfc_main$n } from "./BCol.vue_vue_type_script_setup_true_lang-DrMYlpyB.mjs";
import { _ as _sfc_main$p } from "./BContainer.vue_vue_type_script_setup_true_lang-D1yL7duP.mjs";
import { _ as _sfc_main$1l } from "./BRow.vue_vue_type_script_setup_true_lang-B2V7AEcP.mjs";
import { _ as _sfc_main$q } from "./BDropdown.vue_vue_type_script_setup_true_lang-BT4OzXGx.mjs";
import { _ as _sfc_main$r, a as _sfc_main$s, b as _sfc_main$t, c as _sfc_main$u, d as _sfc_main$v, e as _sfc_main$w, f as _sfc_main$x } from "./BDropdownText.vue_vue_type_script_setup_true_lang-BNJdogga.mjs";
import { _ as _sfc_main$y } from "./BForm.vue_vue_type_script_setup_true_lang-B9irEupU.mjs";
import { _ as _sfc_main$B, a as _sfc_main$D } from "./BFormFloatingLabel.vue_vue_type_script_setup_true_lang-B8Wb_FR9.mjs";
import { _ as _sfc_main$G, a as _sfc_main$K, b as _sfc_main$R, c as _sfc_main$T } from "./BFormValidFeedback.vue_vue_type_script_setup_true_lang-Beu8RDb6.mjs";
import { _ as _sfc_main$z, a as _sfc_main$A } from "./BFormCheckboxGroup.vue_vue_type_script_setup_true_lang-BMu0rSTT.mjs";
import { _ as _sfc_main$C } from "./BFormFile.vue_vue_type_script_setup_true_lang-BybE0gR9.mjs";
import { _ as _sfc_main$E } from "./BFormGroup.vue_vue_type_script_setup_true_lang-DLyoQAHF.mjs";
import { _ as _sfc_main$F } from "./BFormInput.vue_vue_type_script_setup_true_lang-C7CJb3fs.mjs";
import { _ as _sfc_main$H, a as _sfc_main$I } from "./BFormRadioGroup.vue_vue_type_script_setup_true_lang-DyiQkrOF.mjs";
import { _ as _sfc_main$J } from "./BFormRating.vue_vue_type_script_setup_true_lang-9eNNL0_J.mjs";
import { _ as _sfc_main$L, a as _sfc_main$N } from "./BFormSelect.vue_vue_type_script_setup_true_lang-BgFquYjb.mjs";
import { _ as _sfc_main$M } from "./useFormSelect-D2FHF0O9.mjs";
import { _ as _sfc_main$O } from "./BFormSpinbutton.vue_vue_type_script_setup_true_lang-e1TQUYCi.mjs";
import { a as _sfc_main$P, _ as _sfc_main$Q } from "./BFormTags.vue_vue_type_script_setup_true_lang-BRqhTXDR.mjs";
import { _ as _sfc_main$S } from "./BFormTextarea.vue_vue_type_script_setup_true_lang-D6MdvTfG.mjs";
import { _ as _sfc_main$U } from "./BImg.vue_vue_type_script_setup_true_lang-BAusIjYG.mjs";
import { _ as _sfc_main$V, a as _sfc_main$W } from "./BInputGroupText.vue_vue_type_script_setup_true_lang-CnIjPlIE.mjs";
import { _ as _sfc_main$X } from "./BLink.vue_vue_type_script_setup_true_lang-Crq1j_mE.mjs";
import { _ as _sfc_main$Y, a as _sfc_main$Z } from "./BListGroupItem.vue_vue_type_script_setup_true_lang-CLmntYV5.mjs";
import { _ as _sfc_main$_, a as _sfc_main$$ } from "./BModalOrchestrator.vue_vue_type_script_setup_true_lang-4jSwR0hm.mjs";
import { _ as _sfc_main$10, a as _sfc_main$11, b as _sfc_main$12, c as _sfc_main$13, d as _sfc_main$14 } from "./BNavText.vue_vue_type_script_setup_true_lang-DTf8nxUN.mjs";
import { _ as _sfc_main$15, a as _sfc_main$16, b as _sfc_main$17, c as _sfc_main$18 } from "./BNavbarToggle.vue_vue_type_script_setup_true_lang-BKfzvgvU.mjs";
import { _ as _sfc_main$19 } from "./BOffcanvas.vue_vue_type_script_setup_true_lang-CKkvc-Hi.mjs";
import { _ as _sfc_main$1a } from "./BOverlay.vue_vue_type_script_setup_true_lang-BHTZc193.mjs";
import { _ as _sfc_main$1b } from "./BPagination.vue_vue_type_script_setup_true_lang-C_ZVFiFE.mjs";
import { _ as _sfc_main$1c, a as _sfc_main$1d, b as _sfc_main$1e, c as _sfc_main$1f, d as _sfc_main$1g } from "./BPlaceholderWrapper.vue_vue_type_script_setup_true_lang-svTuMwV5.mjs";
import { _ as _sfc_main$1h } from "./floatingUi-BQpU1TSt.mjs";
import { _ as _sfc_main$1i } from "./BPopoverOrchestrator.vue_vue_type_script_setup_true_lang-Co38KaJI.mjs";
import { _ as _sfc_main$1j, a as _sfc_main$1k } from "./BProgress.vue_vue_type_script_setup_true_lang-BX1yVXH1.mjs";
import { _ as _sfc_main$1m } from "./BSpinner.vue_vue_type_script_setup_true_lang-D_xzH5Zg.mjs";
import { _ as _sfc_main$1o, a as _sfc_main$1p, b as _sfc_main$1s, c as _sfc_main$1t, d as _sfc_main$1u, e as _sfc_main$1v, f as _sfc_main$1w, g as _sfc_main$1A } from "./BTable.vue_vue_type_script_setup_true_lang-aBfcsaMN.mjs";
import { _ as _sfc_main$1q } from "./BTableSimple.vue_vue_type_script_setup_true_lang-D2TvxSs4.mjs";
import { _ as _sfc_main$1n, a as _sfc_main$1r } from "./BTabs.vue_vue_type_script_setup_true_lang-CDoJ9hBp.mjs";
import { _ as _sfc_main$1x, a as _sfc_main$1y } from "./BToastOrchestrator.vue_vue_type_script_setup_true_lang-FaJOn6Y0.mjs";
import { _ as _sfc_main$1z } from "./BTooltip.vue_vue_type_script_setup_true_lang-BBsccihF.mjs";
const index = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  BAccordion: _sfc_main,
  BAccordionItem: _sfc_main$1,
  BAlert: _sfc_main$2,
  BAvatar: _sfc_main$3,
  BAvatarGroup: _sfc_main$4,
  BBadge: _sfc_main$5,
  BBreadcrumb: _sfc_main$6,
  BBreadcrumbItem: _sfc_main$7,
  BButton: _sfc_main$8,
  BButtonGroup: _sfc_main$9,
  BButtonToolbar: _sfc_main$a,
  BCard: _sfc_main$b,
  BCardBody: _sfc_main$c,
  BCardFooter: _sfc_main$d,
  BCardGroup: _sfc_main$e,
  BCardHeader: _sfc_main$f,
  BCardImg: _sfc_main$g,
  BCardSubtitle: _sfc_main$h,
  BCardText: _sfc_main$i,
  BCardTitle: _sfc_main$j,
  BCarousel: _sfc_main$k,
  BCarouselSlide: _sfc_main$l,
  BCloseButton: _sfc_main$m,
  BCol: _sfc_main$n,
  BCollapse: _sfc_main$o,
  BContainer: _sfc_main$p,
  BDropdown: _sfc_main$q,
  BDropdownDivider: _sfc_main$r,
  BDropdownForm: _sfc_main$s,
  BDropdownGroup: _sfc_main$t,
  BDropdownHeader: _sfc_main$u,
  BDropdownItem: _sfc_main$v,
  BDropdownItemButton: _sfc_main$w,
  BDropdownText: _sfc_main$x,
  BForm: _sfc_main$y,
  BFormCheckbox: _sfc_main$z,
  BFormCheckboxGroup: _sfc_main$A,
  BFormDatalist: _sfc_main$B,
  BFormFile: _sfc_main$C,
  BFormFloatingLabel: _sfc_main$D,
  BFormGroup: _sfc_main$E,
  BFormInput: _sfc_main$F,
  BFormInvalidFeedback: _sfc_main$G,
  BFormRadio: _sfc_main$H,
  BFormRadioGroup: _sfc_main$I,
  BFormRating: _sfc_main$J,
  BFormRow: _sfc_main$K,
  BFormSelect: _sfc_main$L,
  BFormSelectOption: _sfc_main$M,
  BFormSelectOptionGroup: _sfc_main$N,
  BFormSpinbutton: _sfc_main$O,
  BFormTag: _sfc_main$P,
  BFormTags: _sfc_main$Q,
  BFormText: _sfc_main$R,
  BFormTextarea: _sfc_main$S,
  BFormValidFeedback: _sfc_main$T,
  BImg: _sfc_main$U,
  BInput: _sfc_main$F,
  BInputGroup: _sfc_main$V,
  BInputGroupText: _sfc_main$W,
  BLink: _sfc_main$X,
  BListGroup: _sfc_main$Y,
  BListGroupItem: _sfc_main$Z,
  BModal: _sfc_main$_,
  BModalOrchestrator: _sfc_main$$,
  BNav: _sfc_main$10,
  BNavForm: _sfc_main$11,
  BNavItem: _sfc_main$12,
  BNavItemDropdown: _sfc_main$13,
  BNavText: _sfc_main$14,
  BNavbar: _sfc_main$15,
  BNavbarBrand: _sfc_main$16,
  BNavbarNav: _sfc_main$17,
  BNavbarToggle: _sfc_main$18,
  BOffcanvas: _sfc_main$19,
  BOverlay: _sfc_main$1a,
  BPagination: _sfc_main$1b,
  BPlaceholder: _sfc_main$1c,
  BPlaceholderButton: _sfc_main$1d,
  BPlaceholderCard: _sfc_main$1e,
  BPlaceholderTable: _sfc_main$1f,
  BPlaceholderWrapper: _sfc_main$1g,
  BPopover: _sfc_main$1h,
  BPopoverOrchestrator: _sfc_main$1i,
  BProgress: _sfc_main$1j,
  BProgressBar: _sfc_main$1k,
  BRow: _sfc_main$1l,
  BSpinner: _sfc_main$1m,
  BTab: _sfc_main$1n,
  BTable: _sfc_main$1o,
  BTableLite: _sfc_main$1p,
  BTableSimple: _sfc_main$1q,
  BTabs: _sfc_main$1r,
  BTbody: _sfc_main$1s,
  BTd: _sfc_main$1t,
  BTfoot: _sfc_main$1u,
  BTh: _sfc_main$1v,
  BThead: _sfc_main$1w,
  BToast: _sfc_main$1x,
  BToastOrchestrator: _sfc_main$1y,
  BTooltip: _sfc_main$1z,
  BTr: _sfc_main$1A
}, Symbol.toStringTag, { value: "Module" }));
export {
  index as i
};
